runtime/src/main/java/org/corfudb/runtime/collections/ProtobufIndexer.java

```diff
@@ -49,17 +49,19 @@
 
     private <T> Index.Spec<Message, CorfuRecord<Message, Message>, ?>
     getNestedIndex(String indexPath, String indexName) {
+        // Separate nested fields, as full path is a 'dot' separated String,
+        // e.g., 'person.address.street'
+        String[] nestedFields = indexPath.split("\\.");
```

Comment:
I remember many years ago we had a performance issue with string methods. To avoid those kinds of problems it's recommended to use apache string utils always.

https://stackoverflow.com/questions/11001330/java-split-string-performances

Reply:
This is only called once - per index - on registration.